File: backend/src/main/java/com/afyasmart/backend/service/ReportService.java
package com.afyasmart.backend.service;

/**
 * ============================================================
 * AfyaSmart - Report Service
 * ============================================================
 *
 * Defines the operations available for generating reports.
 *
 * Current reports:
 * - Appointment PDF
 * - Appointment Excel
 * ============================================================
 */
public interface ReportService {

    /**
     * Generates an appointment report in PDF format.
     *
     * @return generated PDF as a byte array
     * @throws Exception if PDF generation fails
     */
    byte[] generateAppointmentPdf() throws Exception;

    /**
     * Generates an appointment report in Excel format.
     *
     * @return generated Excel workbook as a byte array
     * @throws Exception if Excel generation fails
     */
    byte[] generateAppointmentExcel() throws Exception;
}

File: backend/src/main/java/com/afyasmart/frontend/src/services/reportService.js
import api from "./api";

/**
 * ============================================================
 * AfyaSmart - Report Service
 * ============================================================
 *
 * This file contains all API functions related to reports.
 *
 * The backend generates the actual report files.
 * The frontend simply requests the file and downloads it.
 *
 * Current reports:
 *
 * 1. Appointment PDF
 * 2. Appointment Excel
 *
 * ============================================================
 */


/**
 * ============================================================
 * Download Appointment PDF
 * ============================================================
 *
 * Calls:
 *
 * GET /api/reports/appointments/pdf
 *
 * The response is returned as a Blob because a PDF is a
 * binary file rather than normal JSON data.
 */
export const downloadAppointmentPdf = async () => {

    const response = await api.get(
        "/reports/appointments/pdf",
        {
            responseType: "blob"
        }
    );

    return response.data;
};


/**
 * ============================================================
 * Download Appointment Excel
 * ============================================================
 *
 * Calls:
 *
 * GET /api/reports/appointments/excel
 *
 * The response is returned as a Blob because an Excel workbook
 * is a binary file.
 */
export const downloadAppointmentExcel = async () => {

    const response = await api.get(
        "/reports/appointments/excel",
        {
            responseType: "blob"
        }
    );

    return response.data;
};